crates/re_data_store/src/store_read.rs

```diff
@@ -99,108 +93,69 @@ impl RangeQuery {
 
 impl DataStore {
     /// Retrieve all the [`ComponentName`]s that have been written to for a given [`EntityPath`] on
-    /// a specific [`Timeline`].
+    /// the specified [`Timeline`].
     ///
-    /// # Temporal semantics
-    ///
-    /// In addition to the temporal results, this also includes all [`ComponentName`]s present in
-    /// the timeless tables for this entity.
+    /// Static components are always included in the results.
     pub fn all_components(
         &self,
         timeline: &Timeline,
-        ent_path: &EntityPath,
-    ) -> Option<Vec<ComponentName>> {
+        entity_path: &EntityPath,
+    ) -> Option<ComponentNameSet> {
```

Comment:
Not new, but it seems a bit odd to return `None` instead of an empty set.

This construct makes me wonder if `Some(ComponentNameSet::default())` could ever be returned, and how that would differ from `None`

Reply:
It could at least use a comment